src/core/qgsrendercontext.h

```diff
@@ -872,6 +873,29 @@ class CORE_EXPORT QgsRenderContext : public QgsTemporalRangeObject
      */
     void setTextureOrigin( const QPointF &origin );
 
+    /**
+     * Add a painter path to the mask painter path of this render context
+     */
+    void addToMaskPainterPath( QPainterPath const &path ); SIP_SKIP
+
+    /**
+     * Returns the mask painter path stored in this render context
+     * \since QGIS 3.22
+     */
+    QPainterPath maskPainterPath() { return mMaskPainterPath; } SIP_SKIP
```

Comment:
Should this function return a const reference ?

Reply:
Also yes.